Cleanup evicted pods left behind after disk pressure

mattmattox opened the pull request:
When a node starts to evict pods under disk pressure, the evicted pods are left behind. All the resources like volumes, IP, containers, etc will be cleaned up and delete. But the pod object will be left behind in "evicted" status. Per upstream this is [intentional](https://github.com/kubernetes/kubernetes/issues/54525#issuecomment-340035375)

ansilh commented on cleanup-evicted-pods/README.md:
````diff
@@ -0,0 +1,16 @@
+
+### Manual cleanup
+```bash
+kubectl get pods --all-namespaces -ojson | jq -r '.items[] | select(.status.reason!=null) | select(.status.reason | contains("Evicted")) | .metadata.name + " " + .metadata.namespace' | xargs -n2 -l bash -c 'kubectl delete pods $0 --namespace=$1'
````
Do we need `select(.status.reason!=null)` ?

mattmattox replied:
I get the following error if that is removed.

```
> kubectl get pods --all-namespaces -ojson | jq -r '.items[] | select(.status.reason | contains("Evicted")) | .metadata.name + " " + .metadata.namespace'
jq: error (at <stdin>:13385): null (null) and string ("Evicted") cannot have their containment checked
> 
```

ansilh replied:
Thanks, Looks like jq can't loop through null/noexisting objects.

ansilh commented on cleanup-evicted-pods/README.md:
````diff
@@ -0,0 +1,16 @@
+```
+
+### Automatic cleanup
+NOTE: Uses the image `rancher/hyperkube:v1.19.6-rancher1`. This can be changed to match your cluster version by matching the same image tag rke addon jobs in the kube-system.
````
Please add a description stating that it's a cronjob that runs every 30mins and move this note after the description.

mattmattox replied:
Added

ansilh commented on cleanup-evicted-pods/deploy.yaml:
```diff
@@ -0,0 +1,19 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
```
Please add an annotation indicating that it's an OOB Job and not part of RKE or Rancher.

mattmattox replied:
I added a label for `app: cleanup-evicted-pods` is there another annotation I should add?

ansilh replied:
Thanks, @mattmattox . The label is sufficient.

ansilh reviewed (changes requested):
Mac version of xargs won't work . 
Can we mention that the command needs to be executed from a Linux system?
```
xargs: illegal option -- l
usage: xargs [-0opt] [-E eofstr] [-I replstr [-R replacements]] [-J replstr]
             [-L number] [-n number [-x]] [-P maxprocs] [-s size]
             [utility [argument ...]]
```
I've reduced the frequency to one minute during testing and the jobs may overlaps.
Is it make sense to set `concurrencyPolicy` to `Forbid` ? 

```
cleanup-evicted-pods-1612942560-dnmpj   0/1     ContainerCreating   0          36s     <none>           worker03    <none>           <none>
cleanup-evicted-pods-1612942560-ffhd4   0/1     Evicted             0          85s     <none>           worker01    <none>           <none>
cleanup-evicted-pods-1612942620-9j876   0/1     ContainerCreating   0          25s     <none>           worker04    <none>           <none>
```

I've added step to get the image path to use as well as added a note to modify the job after every k8s version upgrade.

mattmattox commented:
@ansilh I switched the image over to rancherlabs/swiss-army-knife so we don't need that section about the image. I also added the concurrencyPolicy as requested.

ansilh reviewed (approved):
Awesome.! :)
